astropy/stats/src/fast_sigma_clip.c

```diff
@@ -105,6 +105,8 @@ static void _sigma_clip_fast(
     double *data_buffer = NULL;
     double *mad_buffer = NULL;
 
+    NPY_BEGIN_THREADS_DEF
```

Comment:
I'm a bit surprised this is necessary; I'd have thought all ufuncs did this by themselves... I'll try to look into this. But I definitely don't want to be in the way of something that works, so no reason to wait for it.

Reply:
It doesn't work at the moment though the tests for sigma clipping basically abort so there is something seriously wrong in this PR. Any insight you have would be appreciated!